src/fileimpl.cpp

```diff
@@ -261,6 +265,14 @@ class Grouping
     m_byTitleDirentLookup.reset(new ByTitleDirentLookup(mp_titleDirentAccessor.get()));
 
     readMimeTypes();
+  } catch (...) {
+    getClusterCache().drop_all([=](const std::tuple<FileImpl*, cluster_index_type>& key) {return std::get<0>(key) == this;});
+    throw;
+  }
+
+  FileImpl::~FileImpl() {
+    // We have to clean the global cache for our clusters.
+    getClusterCache().drop_all([=](const std::tuple<FileImpl*, cluster_index_type>& key) {return std::get<0>(key) == this;});
```

Comment:
Deduplicate `getClusterCache().drop_all(...)` (and make the code more readable/maintainable) by introducing a private function `FileImpl::dropCachedClusters()`.